eam-light-cern/src/main/java/ch.cern.cmms.plugins/SharedPluginImpl.java

```diff
@@ -75,4 +78,17 @@ public Map<String, Map<String, String>> getDatesPermissions(InforClient inforCli
     public String sayHello() {
         return "Hello from CERN!";
     }
+
+    // Retrieve lots from the user defined screen
+    @Override
+    public List<Pair> getUdsLots(String partCode, InforClient inforClient, InforContext context) throws InforException {
+
+        GridRequest udsLotsRequest = new GridRequest("SSPART_U1");
+        udsLotsRequest.addParam("parameter.partcode", partCode);
+
+        GridRequestResult udsLotsResult = inforClient.getGridsService().executeQuery(context, udsLotsRequest);
+
+        return GridTools.convertGridResultToObject(
+                Pair.class, Pair.generateGridPairMap("137670", ""), udsLotsResult);
```

Comment:
Is an empty string as the second argument to generateGridPairMap intended? We don't have a description column?

Reply:
That's right, there is no description

Reply:
Then there is no point in using generateGridPairMap. Create the map yourself and repeat the code as the description

Reply:
Will do

Reply:
I am not seeing how I could repeat the code as the description since that would mean a duplicate key in the map, no? I'd be doing something like:
```java
Map<String, String> lotColumns = new HashMap<String, String>(){{
    put("137670", "code");
    put("137670", "desc"); // duplicate map key
}};

return GridTools.convertGridResultToObject(Pair.class, lotColumns, udsLotsResult);
```

But I might have misunderstood, let me know.